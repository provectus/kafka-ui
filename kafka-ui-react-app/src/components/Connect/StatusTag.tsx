import cx from 'classnames';
import { ConnectorTaskStatus } from 'generated-sources';
import React from 'react';

export interface StatusTagProps {
  status: ConnectorTaskStatus;
}

const StatusTag: React.FC<StatusTagProps> = ({ status }) => {
  const classNames = cx('tag', {
    'is-success': status === ConnectorTaskStatus.RUNNING,
    'is-light': status === ConnectorTaskStatus.PAUSED,
    'is-warning': status === ConnectorTaskStatus.UNASSIGNED,
    'is-danger': status === ConnectorTaskStatus.FAILED,
  });

  return <span className={classNames}>{status}</span>;
};

export default StatusTag;
