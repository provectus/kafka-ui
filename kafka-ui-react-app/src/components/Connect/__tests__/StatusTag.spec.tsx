import React from 'react';
import { create } from 'react-test-renderer';
import StatusTag, { StatusTagProps } from 'components/Connect/StatusTag';
import { ConnectorTaskStatus } from 'generated-sources';

describe('StatusTag', () => {
  const setupWrapper = (props: Partial<StatusTagProps> = {}) => (
    <StatusTag status={ConnectorTaskStatus.RUNNING} {...props} />
  );

  it('matches snapshot for running status', () => {
    const wrapper = create(
      setupWrapper({ status: ConnectorTaskStatus.RUNNING })
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot for failed status', () => {
    const wrapper = create(
      setupWrapper({ status: ConnectorTaskStatus.FAILED })
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot for paused status', () => {
    const wrapper = create(
      setupWrapper({ status: ConnectorTaskStatus.PAUSED })
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot for unassigned status', () => {
    const wrapper = create(
      setupWrapper({ status: ConnectorTaskStatus.UNASSIGNED })
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
