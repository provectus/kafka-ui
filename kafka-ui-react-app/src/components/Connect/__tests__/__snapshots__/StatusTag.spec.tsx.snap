// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusTag matches snapshot for failed status 1`] = `
<span
  className="tag is-danger"
>
  FAILED
</span>
`;

exports[`StatusTag matches snapshot for paused status 1`] = `
<span
  className="tag is-light"
>
  PAUSED
</span>
`;

exports[`StatusTag matches snapshot for running status 1`] = `
<span
  className="tag is-success"
>
  RUNNING
</span>
`;

exports[`StatusTag matches snapshot for unassigned status 1`] = `
<span
  className="tag is-warning"
>
  UNASSIGNED
</span>
`;
